test/integration/consul-container/upgrade/healthcheck_test.go

```diff
@@ -0,0 +1,328 @@
+			ch <- service
+		}
+	}()
+	err = client.Agent().ServiceRegister(&api.AgentServiceRegistration{Name: serviceName, Port: 9998})
+	timer := time.NewTimer(1 * time.Second)
+	select {
+	case err := <-errCh:
+		require.NoError(t, err)
+	case service := <-ch:
+		require.Len(t, service, 1)
+		require.Equal(t, serviceName, service[0].Service.Service)
+		require.Equal(t, 9998, service[0].Service.Port)
+	case <-timer.C:
+		t.Fatalf("test timeout")
+	}
+}
+
+// Test health check GRPC call using Mixed (majority latest) Servers and Latest GA Clients
+func TestMixedServersMajorityLatestGAClient(t *testing.T) {
+	t.Parallel()
+	var configs []consulNode.Config
+	configs = append(configs,
+		consulNode.Config{
+			HCL: `node_name="` + utils.RandName("consul-server") + `"
+					log_level="TRACE"
+					bootstrap_expect=3
+					server=true`,
+			Cmd:     []string{"agent", "-client=0.0.0.0"},
+			Version: *curImage,
+		})
+
+	for i := 1; i < 3; i++ {
+		configs = append(configs,
+			consulNode.Config{
+				HCL: `node_name="` + utils.RandName("consul-server") + `"
+					log_level="TRACE"
+					bootstrap_expect=3
+					server=true`,
+				Cmd:     []string{"agent", "-client=0.0.0.0"},
+				Version: *latestImage,
+			})
+
+	}
+
+	cluster, err := consulCluster.New(configs)
+	require.NoError(t, err)
+	defer Terminate(t, cluster)
+
+	numClients := 1
+	Clients, err := clientsCreate(numClients)
+	client := Clients[0].GetClient()
+	err = cluster.AddNodes(Clients)
+	retry.RunWith(&retry.Timer{Timeout: retryTimeout, Wait: retryFrequency}, t, func(r *retry.R) {
+		leader, err := cluster.Leader()
+		require.NoError(r, err)
+		require.NotEmpty(r, leader)
+		members, err := client.Agent().Members(false)
+		require.Len(r, members, 4)
+	})
+
+	serviceName := "api"
+	err, index := serviceCreate(t, client, serviceName)
+
+	ch := make(chan []*api.ServiceEntry)
+	errCh := make(chan error)
+	go func() {
+		service, q, err := client.Health().Service(serviceName, "", false, &api.QueryOptions{WaitIndex: index})
+		if q.QueryBackend != api.QueryBackendStreaming {
+			err = fmt.Errorf("invalid backend for this test %s", q.QueryBackend)
+		}
+		if err != nil {
+			errCh <- err
+		} else {
+			ch <- service
+		}
+	}()
+	err = client.Agent().ServiceRegister(&api.AgentServiceRegistration{Name: serviceName, Port: 9998})
+	timer := time.NewTimer(1 * time.Second)
+	select {
+	case err := <-errCh:
+		require.NoError(t, err)
+	case service := <-ch:
+		require.Len(t, service, 1)
+		require.Equal(t, serviceName, service[0].Service.Service)
+		require.Equal(t, 9998, service[0].Service.Port)
+	case <-timer.C:
+		t.Fatalf("test timeout")
+	}
+}
+
+// Test health check GRPC call using Mixed (majority current) Servers and Latest GA Clients
+func TestMixedServersMajorityCurrentGAClient(t *testing.T) {
+	t.Parallel()
+	var configs []consulNode.Config
+	configs = append(configs,
+		consulNode.Config{
+			HCL: `node_name="` + utils.RandName("consul-server") + `"
+					log_level="TRACE"
+					bootstrap_expect=3
```

Comment:
```suggestion
```

I would remove this to prevent this node from gaining leadership accidentally. It appears like this scenario is trying to mimic the one where new servers are being added to the cluster to perform an upgrade but there are not enough of the newer ones there yet. In that case leaving out the bootstrap expect ensures this node doesn't accidentally gain first leadership when the cluster is bootstrapped and therefore will more accurately mimic the desired state.